WooCommerce/WooCommerceTests/ViewRelated/Products/Media/ProductImageUploaderTests.swift

```diff
@@ -693,6 +729,265 @@ final class ProductImageUploaderTests: XCTestCase {
         imageUploader.sendBackgroundUploadNoticeIfNeeded(key: key, using: noticePresenter)
         XCTAssertTrue(isNoticeTriggered)
     }
+
+    // MARK: - Tests with background image upload feature flag enabled
+
+    func test_hasUnsavedChangesOnImages_becomes_false_after_uploading_and_saving_with_flag_enabled() {
+        // Given
+        mockFeatureFlagService = MockFeatureFlagService(backgroundProductImageUpload: true)
+        let mockImageUploader = MockProductImageUploader()
+        let key = ProductImageUploaderKey(siteID: siteID,
+                                         productOrVariationID: productID,
+                                         isLocalID: false)
+
+        // When
+        mockImageUploader.whenHasUnsavedChangesOnImagesIsCalled(thenReturn: true)
+        XCTAssertTrue(mockImageUploader.hasUnsavedChangesOnImages(key: key, originalImages: []))
+
+        // Then
+        mockImageUploader.whenHasUnsavedChangesOnImagesIsCalled(thenReturn: false)
+        XCTAssertFalse(mockImageUploader.hasUnsavedChangesOnImages(key: key, originalImages: [.fake().copy(imageID: 123)]))
+    }
```

Comment:
I think this test is redundant as you were testing the mock instead of the actual `ProductImageUploader`.

Reply:
Yes and I agree, this is pretty similar to the one with the feature flag disabled, so I rewrote it using the actual `ProductImageUploader` implementation with the feature flag enabled 0d4be44c1bed58695eed6fa2cf32a7f87dc21831 

Here is what I do:
1. Verifying initial state has no unsaved changes
2. Uploading an image and confirming it creates unsaved changes
3. Saving the product and confirming changes are now saved

Let me know what you think.

Reply:
Thanks! LGTM 👍